lib/dnsimple/client.rb

```diff
@@ -70,53 +70,57 @@ def base_url
     # Make a HTTP GET request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Query and header params for request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
     def get(path, options = {})
-      execute :get, path, options
+      execute :get, path, {}, options
     end
 
     # Make a HTTP POST request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Body and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
-    def post(path, options = {})
-      execute :post, path, options
+    def post(path, data, options= {})
+      execute :post, path, data, options
     end
 
     # Make a HTTP PUT request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Body and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
-    def put(path, options = {})
-      execute :put, path, options
+    def put(path, data, options = {})
+      execute :put, path, data, options
     end
 
     # Make a HTTP PATCH request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Body and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
-    def patch(path, options = {})
-      execute :patch, path, options
+    def patch(path, data, options = {})
+      execute :patch, path, data, options
     end
 
     # Make a HTTP DELETE request.
     #
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Query and header params for request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
     def delete(path, options = {})
-      execute :delete, path, options
+      execute :delete, path, {}, options
```

Comment:
Although the RFC doesn't mention the data payload for a `DELETE` action, it's common practice to consider `DELETE` to accept a payload. I would allow it, since I believe we use (or used) it.